docker/docker.ts

```diff
@@ -351,36 +376,27 @@ async function pushImageAsync(
     tag = tag ? `:${tag}` : "";
     const targetImage = `${repositoryUrl}${tag}`;
 
-    const tagResult = await runCLICommand(
-        "docker", ["tag", imageName, targetImage], logResource, /*reportStdout*/ true);
-
-    if (tagResult.code) {
-        throw new ResourceError(`Failed to tag Docker image with remote registry URL ${repositoryUrl}`, logResource);
-    }
-    const pushResult = await runCLICommand("docker", ["push", targetImage], logResource, /*reportStdout*/ true);
-    if (pushResult.code) {
-        throw new ResourceError(`Docker push of image '${imageName}' failed with exit code: ${pushResult.code}`, logResource);
-    }
+    await runCommandThatMustSucceed("docker", ["tag", imageName, targetImage], logResource);
+    await runCommandThatMustSucceed("docker", ["push", targetImage], logResource);
 }
 
 // getDigest returns the digest, if available, of a target image.  The digest will only be available once the repo image
 // is pushed or pulled from the target repository.
 export async function getDigest(targetImage: string, logResource: pulumi.Resource): Promise<string | undefined> {
     // Do not forward the output of this command this to the CLI to show the user.
 
-    const inspectResult = await runCLICommand(
-        "docker", ["image", "inspect", targetImage], logResource, /*reportStdout*/ false);
-    if (inspectResult.code || !inspectResult.stdout) {
+    const inspectResult = await runCommandThatMustSucceed("docker", ["image", "inspect", targetImage], logResource);
+    if (!inspectResult) {
         throw new ResourceError(
-            `No digest available for image ${targetImage}: ${inspectResult.code} -- ${inspectResult.stdout}`, logResource);
+            `No digest available for image ${targetImage}`, logResource);
```

Comment:
only need to examine the stdout results, no need to check the code as runCommandThatMustSucceed does that for you.